pkg/odo/cli/config/unset.go

```diff
@@ -56,6 +58,12 @@ func (o *UnsetOptions) Complete(name string, cmd *cobra.Command, args []string)
 	if o.envArray == nil {
 		o.paramName = args[0]
 	}
+	if len(o.componentContext) > 0 {
```

Comment:
validation should be in `Validate` function

Reply:
resolved